### PR TITLE
Set partition to disjoint set

### PR DESCRIPTION
**Pull request by thecaligarmo**
This issue fixes the bug in #38415 . It uses the base of `SetPartition` to allow a `DisjointSet` to be created in a nice way. This also solves issue #38416 

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

#38415 , #38416

### CODE REVIEW

File: src/sage/sets/disjoint_set.pyx [hunk 1]
```diff
@@ -103,6 +104,15 @@ cpdef DisjointSet(arg):
...
         {{'cheval'}, {'yi'}, {45}}
 
+    From a set partition::
+
+        sage: SP = SetPartition(DisjointSet(5))
+        sage: DisjointSet(SP)
+        {{0}, {1}, {2}, {3}, {4}}
+        sage: DisjointSet(SP) == DisjointSet(5)
+        True
+
```
Comment (dcoudert) [inline src/sage/sets/disjoint_set.pyx:114]: 1 empty line is enough

File: src/sage/sets/disjoint_set.pyx [hunk 2]
```diff
@@ -103,6 +104,15 @@ cpdef DisjointSet(arg):
         sage: DisjointSet(['yi', 45, 'cheval'])
         {{'cheval'}, {'yi'}, {45}}
 
+    From a set partition::
```
Comment (dcoudert) [inline src/sage/sets/disjoint_set.pyx]: you could add a reference to the issue or the PR using ```(see :issue:`12345`)```

**Review by dcoudert — COMMENTED** [review]
otherwise LGTM.

**Comment by thecaligarmo** [comment]
Fixed. I also forgot that this issue handles #38416 by providing the alternative solution.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Review by dcoudert — APPROVED** [review]
go back to positive review after rebase

**Pull request merged**